packages/core/src/provider/base.ts

```diff
@@ -540,4 +562,33 @@ export class SmartAccountProvider<
       };
     };
   };
+
+  private _runMiddlewareStack = async (
+    data: UserOperationRequest,
+    overrides?: UserOperationOverrides
+  ) => {
+    if (!this.account) {
+      throw new Error("account not connected!");
+    }
+
+    const uoStruct = await asyncPipe(
+      this.dummyPaymasterDataMiddleware,
+      this.feeDataGetter,
+      this.gasEstimator,
+      this.paymasterDataMiddleware,
+      this.customMiddleware ?? noOpMiddleware,
+      // This applies the overrides if they've been passed in
+      async (struct) => ({ ...struct, ...overrides })
```

Comment:
The paymaster error makes sense, and is something I totally overlooked. These actually need to run before the paymaster middleware runs.

It also means that we can't run the full stack during drop and replace.

It needs to run as:
1. get fees
2. Apply drop and replace logic
3. Run the rest of the stack
4. Send user op

Reply:
Lemme get back to you on this tomorrow or Thursday, might need to rethink the approach here slightly

Reply:
Can we say that the paymasterMiddleware should always be at the end?

I am not sure what kind of middlewares devs will use as `customMiddleware` but the process of signature should be at the end right?